texar/modules/encoders/xlnet_encoders.py

```diff
@@ -0,0 +1,479 @@
+            Hidden dimension of the embeddings
+
+        "head_dim": int
+            Size of the vectors after head projection.
+
+        "dropout": float
+            Dropout rate for layers
+
+        "attention_dropout": floar
+            Dropout rate for attention layers
+
+        "use_segments": bool
+            Boolean to indicate if the input has segments
+
+        "ffn_inner_dim": int
+            Dimension of PositionWise FF network's hidden layer
+
+        "activation": str or callable
+            Activation function applied to the output of the PositionWise FF.
+            See :func:`~texar.core.get_activation_fn` for more details.
+
+        "vocab_size" : int
+            The vocabulary size of `inputs` in `XLNet`.
+
+        "max_seq_len": int
+            Maximum len of the sequence allowed in one segment
+
+        "name" : str
+            Name of the module.
+        """
+
+        return {
+            "name": "xlnet_encoder",
+            'pretrained_model_name': 'xlnet-base-cased',
+            "untie_r": True,
+            "num_layers": 12,
+            "mem_len": 0,
+            "reuse_len": 0,
+            # initializer
+            "initializer": None,
+            # layer
+            "num_heads": 12,
+            "hidden_dim": 768,
+            "head_dim": 64,
+            "dropout": 0.1,
+            "attention_dropout": 0.1,
+            "use_segments": True,
+            # ffn
+            "ffn_inner_dim": 3072,
+            "activation": 'gelu',
+            # embedding
+            "vocab_size": 32000,
+            "max_seq_len": 512,
+            '@no_typecheck': ['pretrained_model_name']
+        }
+
+    @property
+    def output_size(self):
+        r"""Return the output size of the network.
+        """
+        return self._hparams.hidden_dim
+
+    @staticmethod
+    def _cache_mem(curr_out, prev_mem, mem_len, reuse_len=None):
+        r"""Cache hidden states into memory."""
+        assert mem_len > 0
+
+        if reuse_len is not None and reuse_len > 0:
+            curr_out = curr_out[:reuse_len]
+
+        if prev_mem is None:
+            new_mem = curr_out[-mem_len:]
+        else:
+            new_mem = tf.concat([prev_mem, curr_out], 0)[-mem_len:]
+
+        return tf.stop_gradient(new_mem)
+
+    def _create_mask(self, qlen, mlen, dtype=tf.float32, same_length=False):
+        r"""Create causal attention mask."""
+        attn_mask = tf.ones([qlen, qlen], dtype=dtype)
+        mask_u = tf.matrix_band_part(attn_mask, 0, -1)
+        mask_dia = tf.matrix_band_part(attn_mask, 0, 0)
+        attn_mask_pad = tf.zeros([qlen, mlen], dtype=dtype)
+        ret = tf.concat([attn_mask_pad, mask_u - mask_dia], axis=1)
+        if same_length:
+            mask_l = tf.matrix_band_part(attn_mask, -1, 0)
+            ret = tf.concat([ret[:, :qlen] + mask_l - mask_dia, ret[:, qlen:]],
+                            axis=1)
+
+        return ret
+
+    def _build(self, inputs, *args, **kwargs):
```

Comment:
`_build` is exposed to users while `_execute` is not. Pls move `_execute`'s docstring to `_build`. Users are not aware of the existence of `_execute`. Take the pytorch version as a reference.

Reply:
Done! [9280366]